common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -555,6 +583,19 @@ def should_force_account_creation():
 
 
 @partial.partial
+def create_user(strategy, details, user=None, *args, **kwargs):
+    """
+    Substitution method for stock social create_user that catches email validation error and redirects to login
+    """
+    from student.views import AccountEmailAlreadyExistsValidationError
+    try:
+        return social_create_user(strategy, details, user, *args, **kwargs)
+    except AccountEmailAlreadyExistsValidationError as exc:
+        logger.exception(exc.message)
+        raise EmailAlreadyInUseException(exc.message, details['email'])
```

Comment:
This code is a little confusing - at first glance it makes no sense that you are just catching and re-raising an almost identical exception. Can you please add a little comment/docstring notice explaining that the point is to raise an exception that inherits from `AuthException` ?